static/js/publisher/pages/RegisterSnap/RegisterSnapForm.tsx

```diff
@@ -0,0 +1,152 @@
+import { ReactNode, useState, Dispatch, SetStateAction } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import {
+  Form,
+  Select,
+  Input,
+  Button,
+  Icon,
+  Row,
+  Col,
+} from "@canonical/react-components";
+
+import type { RegistrationResponse } from "./local-types";
+
+type Props = {
+  isSending: boolean;
+  setIsSending: Dispatch<SetStateAction<boolean>>;
+  setRegistrationResponse: Dispatch<
+    SetStateAction<RegistrationResponse | undefined>
+  >;
+  availableStores: { name: string; id: string }[];
+};
+
+function RegisterSnapForm({
+  isSending,
+  setIsSending,
+  setRegistrationResponse,
+  availableStores,
+}: Props): ReactNode {
+  const [snapName, setSnapName] = useState<string>();
+  const [selectedStore, setSelectedStore] = useState<string>("ubuntu");
+  const [isPrivate, setIsPrivate] = useState<string>("private");
+
+  const navigate = useNavigate();
+
+  return (
+    <Form
+      onSubmit={async (e) => {
```

Comment:
That's quite a lot of code to be put in an inline `onSubmit` handler. Could this be extracted to a function?

Reply:
Good point. I'll extract it.